File: LeetCode/cpp/2299_Strong_Password_Checker_II.cpp
class Solution {
public:
	bool containsLowercase(string password) {
		for (int i = 0; i < password.length(); i++) {
			if (password[i] >= 'a' && password[i] <= 'z') {
				return true;
			}
		}

		return false;
	}

	bool containsUppercase(string password) {
		for (int i = 0; i < password.length(); i++) {
			if (password[i] >= 'A' && password[i] <= 'Z') {
				return true;
			}
		}

		return false;
	}

	bool containsDigit(string password) {
		for (int i = 0; i < password.length(); i++) {
			if (password[i] >= '0' && password[i] <= '9') {
				return true;
			}
		}

		return false;
	}

	bool containsSpecialCharacter(string password) {
		string sp = "!@#$%^&*()-+";
		set<char> st;
		for (int i = 0; i < sp.length(); i++) {
			st.insert(sp[i]);
		}

		for (int i = 0; i < password.length(); i++) {
			if (st.find(password[i]) != st.end()) {
				return true;
			}
		}

		return false;
	}

	bool strongPasswordCheckerII(string password) {
		if (password.length() < 8) {
			return false;
		}

		for (int i = 0; i < password.length() - 1; i++) {
			if (password[i] == password[i + 1]) {
				return false;
			}
		}

		return containsLowercase(password) && containsUppercase(password) && containsDigit(password) && containsSpecialCharacter(password);
	}
};

File: LeetCode/java/2299_Strong_Password_Checker_II.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    private boolean containsLowercase(String password) {
        for (int i = 0; i < password.length(); i++) {
            if (password.charAt(i) >= 'a' && password.charAt(i) <= 'z') {
                return true;
            }
        }

        return false;
    }

    private boolean containsUppercase(String password) {
        for (int i = 0; i < password.length(); i++) {
            if (password.charAt(i) >= 'A' && password.charAt(i) <= 'Z') {
                return true;
            }
        }

        return false;
    }

    private boolean containsDigit(String password) {
        for (int i = 0; i < password.length(); i++) {
            if (password.charAt(i) >= '0' && password.charAt(i) <= '9') {
                return true;
            }
        }

        return false;
    }

    private boolean containsSpecialCharacter(String password) {
        String sp = "!@#$%^&*()-+";
        Set<Character> set = new HashSet<>();
        for (int i = 0; i < sp.length(); i++) {
            set.add(sp.charAt(i));
        }

        for (int i = 0; i < password.length(); i++) {
            if (set.contains(password.charAt(i))) {
                return true;
            }
        }

        return false;
    }

    public boolean strongPasswordCheckerII(String password) {
        if (password.length() < 8) {
            return false;
        }

        for (int i = 0; i < password.length() - 1; i++) {
            if (password.charAt(i) == password.charAt(i + 1)) {
                return false;
            }
        }

        return containsLowercase(password) && containsUppercase(password) && containsDigit(password) && containsSpecialCharacter(password);
    }
}